cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -364,6 +365,8 @@ protected Future<Void> reconcileConnectors(Reconciliation reconciliation, T conn
             connectStatus.setConnectorPlugins(connectorPlugins);
 
             connectorsResourceCounter(namespace).set(desiredConnectors.size());
+            long pausedConnectorsCount = desiredConnectors.stream().filter(Annotations::isReconciliationPausedWithAnnotation).count();
+            pausedConnectorsResourceCounter(namespace).set((int) pausedConnectorsCount);
```

Comment:
TBH, looking at the code, I wonder if the metrics work at all. It seems to count the number of connectors per-Connect instance. But there might be multiple instances of Connect per namespace which I guess will be overwriting each others counts? Is that something you tried while working on this by any chance?

Reply:
Yeah, it doesn't work. The same problem with `strimzi_resources{kind="KafkaConnector"` metric.

Reply:
Aggregation could be easily done if it's possible to extend `MetricsProvider` interface. But probably you have only AtomicInteger-based gauge for some reason.
That is how it could be looks like: https://github.com/ortex/strimzi-kafka-operator/commit/60ff7378a970abb2d64a7f0b1aa0ba8c3f799ffb

I didn't add this commit to PR because not sure if we go with this.

Btw, with per-connect approach connectors with the wrong `strimzi.io/cluster` label don't count. Perhaps could be fixed.

Anyway, I'm going to look more deeply at the watcher and how per-connector counting could be done.

Reply:
That is an interesting approach. I do not remember any specific reason why we have only the Atomic Integer. I would say it was all we needed. TBH, I didn't even know you can aggregate the metric like you do it in your branch.

I had a brief look at the connector watch. It would be easy to fix the total resource counter there. You can just increment / decrement the integer when the watch is called for created / deleted connectors. But it would not be completely easy to handle the paused connectors since it will make it hard to handle paused / unpaused connectors to increment / decrement the counter. So I guess your approach is better.

In theory, I guess one can also combine it? Count the total resources in the watch (that way it will include all resources even with wrong `strimzi.io/cluster` label) but the paused resources only in the Connect reconciliation? The connectors with wrong label are not really reconciled anyway, so the paused / unpaused state doesn't make much sense anyway.

Reply:
Yeah, I'll try a combined approach